test_conformance/integer_ops/test_extended_bit_ops_insert.cpp

```diff
@@ -0,0 +1,207 @@
+//
+// Copyright (c) 2021 The Khronos Group Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <algorithm>
+#include <numeric>
+#include <string>
+#include <vector>
+
+#include "procs.h"
+#include "integer_ops_test_info.h"
+#include "harness/testHarness.h"
+
+template <typename T>
+static T cpu_bit_insert(T tbase, T tinsert, cl_uint offset, cl_uint count)
+{
+    assert(offset <= sizeof(T) * 8);
+    assert(count <= sizeof(T) * 8);
+    assert(offset + count <= sizeof(T) * 8);
+
+    cl_ulong base = static_cast<cl_ulong>(tbase);
+    cl_ulong insert = static_cast<cl_ulong>(tinsert);
+
+    cl_ulong mask = (count < 64) ? ((1ULL << count) - 1) << offset : ~0ULL;
+    cl_ulong result = ((insert << offset) & mask) | (base & ~mask);
+
+    return static_cast<T>(result);
```

Comment:
This should be rewritten so as not to rely on implementation-defined behavior for negative values.

Reply:
Fixed by computing the reference results as unsigned types.

Reply:
Looks good to me.